config/routes.rb

```diff
@@ -0,0 +1,7 @@
+Rails.application.routes.draw do
+  root "mains#index"
+  get 'users/', to: 'users#index'
+  get 'users/:user_id', to: 'users#show'
+  get 'users/:user_id/posts', to: 'posts#index'
+  get 'users/:user_id/posts/:id', to: 'posts#show'
```

Comment:
[OPTIONAL] You can use the Rails Default Resource Routing, this will help you have your code cleaner because will be easier to understand and read.

```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Thanks for the review @petudeveloper and @enaburekhan